src/test/test_switches.cpp

```diff
@@ -0,0 +1,315 @@
+/**
+ * This file is part of ExpressLRS
+ * See https://github.com/AlessandroAU/ExpressLRS
+ *
+ * Unit tests for over the air packet encoding, decoding and associated utilities
+ *
+ * Entry point is setup_switches()
+ */
+
+#include <Arduino.h>
+#include <unity.h>
+
+#include "LoRaRadioLib.h"
+#include "CRSF.h"       // has to come after LoraRadioLib.h for R9
+#include "targets.h"
+#include "common.h"
+#include <OTA.h>
+
+// Tests have only been run on TTGO V1, so we have to guard
+// all the testcode to prevent breakages until it's know to do
+// something sensible.
+
+#ifdef TARGET_TTGO_LORA_V1_AS_TX
+
+CRSF crsf;  // need an instance to provide the fields used by the code under test
+
+SX127xDriver Radio; // needed for Radio.TXdataBuffer
+
+/* Check that the round robin works
+ * First call should return 0 for seq switches or 1 for hybrid
+ * Successive calls should increment the next index until wrap
+ * around from 7 to either 0 or 1 depending on mode.
+ */
+void test_round_robin(void) {
+
+    uint8_t expectedIndex=crsf.nextSwitchIndex;
+
+    for(uint8_t i=0; i<10; i++) {
+        uint8_t nsi = crsf.getNextSwitchIndex();
+        TEST_ASSERT_EQUAL(expectedIndex, nsi);
+        expectedIndex++;
+        if (expectedIndex == 8) {
+#ifdef HYBRID_SWITCHES_8
+            expectedIndex = 1;
+#else
+            expectedIndex = 0;
+#endif
+        }
+    }
+}
+
+/* Check that a changed switch gets priority
+*/
+void test_priority(void) {
```

Comment:
Curly brace should be on the next line here

Reply:
I disagree, but I changed it for you anyway.

Reply:
Happy for my formatting requests to be challenged, but it was inconsistent with the other function headers in the file ;)